src/API/Notes.php

```diff
@@ -227,6 +240,28 @@ public function update_item( $request ) {
 		return $this->get_item( $request );
 	}
 
+	/**
+	 * Undo delete all notes.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Request|WP_Error
+	 */
+	public function undoremove_items( $request ) {
+		$data      = array();
+		$notes_ids = $request->get_param( 'notesIds' );
+		foreach ( (array) $notes_ids as $note_id ) {
+			$note = WC_Admin_Notes::get_note( $note_id );
```

Comment:
Should we cast $note_id to an (int) just to be extra safe?

Reply:
Good idea! Sounds good to me. I added this to the commit [926424c](https://github.com/woocommerce/woocommerce-admin/pull/4281/commits/926424cb334572f79d87a89d88e358ffc21f185d).